core/src/main/java/io/questdb/cutlass/line/udp/LineUdpParserSupport.java

```diff
@@ -0,0 +1,302 @@
+                        row.putChar(columnIndex, value.length() == 2 ? (char) 0 : value.charAt(1)); // skip quotes
+                        break;
+                    case ColumnType.GEOBYTE:
+                        row.putByte(
+                                columnIndex,  // skip quotes
+                                (byte) GeoHashes.fromStringTruncatingNl(
+                                        value,
+                                        1,
+                                        value.length() - 1,
+                                        columnTypeMeta
+                                )
+                        );
+                        break;
+                    case ColumnType.GEOSHORT:
+                        row.putShort(
+                                columnIndex,
+                                (short) GeoHashes.fromStringTruncatingNl(
+                                        value,
+                                        1,
+                                        value.length() - 1,
+                                        columnTypeMeta
+                                )
+                        );
+                        break;
+                    case ColumnType.GEOINT:
+                        row.putInt(
+                                columnIndex,
+                                (int) GeoHashes.fromStringTruncatingNl(
+                                        value,
+                                        1,
+                                        value.length() - 1,
+                                        columnTypeMeta
+                                )
+                        );
+                        break;
+                    case ColumnType.GEOLONG:
+                        row.putLong(
+                                columnIndex,
+                                GeoHashes.fromStringTruncatingNl(
+                                        value,
+                                        1,
+                                        value.length() - 1,
+                                        columnTypeMeta
+                                )
+                        );
+                        break;
+                    case ColumnType.NULL:
+                    default:
+                        // unsupported types and null are ignored
+                        break;
+                }
+            } catch (NumericException e) {
+                LOG.info()
+                        .$("cast error [value=")
+                        .$(value).$(", toType=")
+                        .$(ColumnType.nameOf(columnType))
+                        .$(']')
+                        .$();
+            }
+        } else {
+            putNullValue(row, columnIndex, columnType);
+        }
+    }
+
+    public static class BadCastException extends Exception {
+        public static final BadCastException INSTANCE = new BadCastException();
+    }
+
+    public static int getValueType(CharSequence value) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 25. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.